docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,586 @@
+example:
+
+```cvl
+balanceOf(address) returns(uint) envfree
+```
+will become
+```cvl
+function balanceOf(address) external returns(uint) envfree;
+```
+(note also the addition of `external`, see {ref}`described below <cvl2-visibility>`).
+
+This is also true for entries with summaries:
+```cvl
+_setManagedBalance(address,uint256) => NONDET
+```
+will become
+```cvl
+function _setManagedBalance(address,uint256) internal => NONDET;
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Required `;` in more places
+
+`using`, `pragma`, `import`, and `use` statements all require a `;` at the end.  For
+example,
+
+```cvl
+using C as c
+```
+
+becomes
+```cvl
+using C as c;
+```
+
+Note: `use` statements do not require (and may not have) a semicolon if they
+are followed by a `preserved` or `filtered` block.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Method literals require `sig:`
+
+In some places in CVL, you can refer to a contract method by its name and
+argument types.  For example, you might write
+```cvl
+require f.selector == balanceOf(address).selector;
+```
+
+In this example, `balanceOf(address)` is a *method literal*.  In CVL 2,
+these methods literals must now start with `sig:`.  For example, the above
+would become:
+
+```cvl
+require f.selector == sig:balanceOf(address).selector;
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Stricter ordering on method annotations
+
+In CVL 2, the order of the annotations must be visibility modifiers (`internal` or `external`),
+followed by `returns` clause (if any), followed by `optional`, `library`, or `envfree` in any order (if any),
+followed by a summary (if any).
+
+CVL 1 was less strict about the order.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Use of contract name instead of `using` variable
+
+In CVL 1, the only way to refer to a contract on the scene was to first
+introduce a variable with a `using` statement, and then use that variable.  For
+example, to access a struct type `S` defined in `Example.sol`, you would need
+to write
+
+```cvl
+using Example as c;
+
+rule example {
+    c.S x = getAnS();
+}
+```
+
+In CVL 2, you must now use the name of the contract, rather than the variable,
+when referring to user-defined types.  The above example would now be written
+
+```cvl
+rule example {
+    Example.S x = getAnS();
+}
+```
+
+There is no need for a `using` statement in this example.
+
+`using` statements are still required to call methods on secondary contracts.
+
+```{todo}
+Error message
+```
+
+Changes to methods block entries
+--------------------------------
+
+In addition to the superficial changes listed above, there are some changes that
+change the way that methods block entries can be written.  In CVL 1, `methods`
+block entries often had several different functions and meanings:
```

Comment:
```suggestion
block entries often have several different functions and meanings:
```
Changed to present tense since CVL 1 still exists and the points immediately below are in present tense.

Reply:
I decided to change them all to past tense because I think that's what the rest of the document does